src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp

```diff
@@ -246,13 +244,33 @@ struct Charge : db::SimpleTag {
 };
 
 /*!
- * \brief The mass of the particle.
+ * \brief Whether to apply the acceleration due to the scalar self-force to the
+ * particle
+ */
+struct ApplySelfForce : db::SimpleTag {
+  using type = bool;
+  using option_tags = tmpl::list<OptionTags::SelfForceOptions>;
+  static constexpr bool pass_metavariables = false;
+  static bool create_from_options(
+      const std::optional<OptionTags::SelfForceOptions>& self_force_options) {
+    return self_force_options.has_value();
+  };
+};
+
+/*!
+ * \brief The mass of the scalar charge. Only has a value is `ApplySelfForce` is
+ * true.
  */
 struct Mass : db::SimpleTag {
-  using type = double;
-  using option_tags = tmpl::list<OptionTags::Mass>;
+  using type = std::optional<double>;
```

Comment:
An idea to maybe avoid having to hold individual optionals in tags for self force things and check if they are available: You could just store the `SelfForceOptions` that were constructed from the input file in a tag. Then that tag can be used instead of `ApplySelfForce` (you just check the optional), and also wherever you need it's parameters. Thoughts?

Reply:
I removed `ApplySelfForce` now in favor of just checking whether `MaxIterations` is greater than 0. I think I would prefer to hide the extraction of these quantities in the OptionParsing, rather than having to get them from the struct each time step.